### PR TITLE
asio: add old version 1.10.8

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Asio v1.10.8 need to build MariaDB Galera plugins

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/asio/generic.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{stdenv, fetchurl, boost, openssl
+, version, sha256, ...
```
Comment (veprbl) [inline pkgs/development/libraries/asio/generic.nix]: Why do you need triple dots here?

Comment (Izorkin) [reply pkgs/development/libraries/asio/generic.nix]: Not work without triple dots

nix build -f /home/test/src_nix/nixpkgs_my asio_1_10                                                                                                                     
error: anonymous function at /home/test/src_nix/nixpkgs_my/pkgs/development/libraries/asio/generic.nix:1:1 called with unexpected argument 'callPackage', at /home/test/src_nix/nixpkgs_my/lib/customisation.nix:74:12

**Comment by xeji** [comment]
@GrahamcOfBorg  build asio_1_10 asio_1_12

**Comment by xeji** [comment]
No, previous versions built on Darwin: https://hydra.nixos.org/job/nixpkgs/trunk/asio.x86_64-darwin
but have failed since 1.12.1. Not related to this PR.

**Pull request merged**